### ISSUE TITLE
2019-12-05 M3SS fails with tc103, M1SS with tc011, M1SS with tc012, M3SS with tc016 with parameterset=="PARAMETERDISPLAYLIST"

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#84]
tc103 was executed in the openNCA application with the attached data, mct and flags with the following error message: (note that flags and mct are in JSON format, csv format will be provided as an update)

This is a critical issue and is related to setting parameterset=="PARAMETERDISPLAYLIST" rather than "PARAMETERLIST". See below

```
There were 25 warnings (use warnings() to see them)
[1] "openNCA: Model: M3 DosingType: SS AUCMETHOD: LINLOG"
Error in value[[3L]](cond) : 
Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 216997173
Calls: run_computation ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
In addition: Warning message:
In run_M3_SS_computation(data = merged_data, map = map_data, method = method, :
Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
Execution halted
```

It's very clear that parameterset=="PARAMETERDISPLAYLIST" is NO longer being respected in the computation engine. Note the following example: It would appear here that the create_dependency_list approach with comp_required/disp_required approaches are no longer effective in controlling dependencies. Here CEST is being computed but CEST is not required for any of the parameters specified in the PARAMETERDISPLAYLIST. It's possible that CEST is required for the population of the estimated concentration dataset but it's unexpected that TLAST is failing here. 

```
 parameterset
[1] "PARAMETERDISPLAYLIST"
> mct$PARAMETERDISPLAYLIST
[1] "CMAX;KEL;KELNOPT;KELRSQ;KELTMHI;KELTMLO;THALF"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in cest(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 49420991
In addition: Warning messages:
1: In run_M3_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
2: In run_M3_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
3: In run_M3_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in cest(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 49420991 
```

## Example with M1SS - tc008 M1SS fails as well with parameterset set to "PARAMETERDISPLAYLIST"

parameterset == "PARAMETERDISPLAYLIST"
```
> (tclabel <- paste0("results.", testcase.id, ".", mct$MODEL, ".", mct$DOSINGTYPE, ".", mct$AUCMETHOD))
[1] "results.tc008.M1.SS.LINLOG"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 205758774
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTO'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 205758774
```

## tc011 M1SS fails with parameterset=="PARAMETERDISPLAYLIST"

```
> parameterset
[1] "PARAMETERDISPLAYLIST"
> mct$PARAMETERDISPLAYLIST
[1] "SUBJID;TREATXT;PKBDFLD;PKTERM;PERIODU;PERIOD;VISITU;VISIT;DOSE;DOSEU;ACTTRT;AUCTAU;AUCLASTi;AUCXPCTP;CLFTAUi;CMAXi;KEL;KELNOPT;KELRSQ;KELTMHI;KELTMLO;THALF;TLAST;TMAXi;VZFP;CAV;CTROUGH;CMIN;PTR;PTF;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;FLGACCEPTTAU;THALFF"
mct$AUCMETHOD))
[1] "results.tc011.M1.SS.LINLOG"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 70745987
In addition: Warning message:
In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 70745987 
```

## tc012 M1SS fails with parameterset=="PARAMETERDISPLAYLIST"

```
> parameterset
[1] "PARAMETERDISPLAYLIST"
> mct$PARAMETERDISPLAYLIST
[1] "SUBJID;TREATXT;PKBDFLD;PKTERM;PERIODU;PERIOD;VISITU;VISIT;DOSE;DOSEU;ACTTRT;AUCTAU;AUCLASTi;AUCXPCTP;CLFTAUi;CMAX;CMAXi;KEL;KELNOPT;KELRSQ;KELTMHI;KELTMLO;THALF;TLAST;TMAX;TMAXi;VZFP;CAV;CTROUGH;CMIN;PTR;PTF;RACC;RSS;RACCMAXi;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;FLGACCEPTTAU;THALFF"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 24280050
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTO'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 24280050 
```

## tc016 M3SS fails with parameterset=="PARAMETERDISPLAYLIST"

```
> parameterset
[1] "PARAMETERDISPLAYLIST"
> mct$PARAMETERDISPLAYLIST
[1] "SUBJID;TREATXT;PKBDFLD;PKTERM;PERIODU;PERIOD;VISITU;VISIT;DOSE;DOSEU;ACTTRT;AUCTAUi;AUCLASTi;AUCXPCTP;CMAXi;CAVi;KEL;KELRSQ;KELNOPT;KELRSQ;KELTMHI;KELTMLO;THALF;TLAST;TMAXi;MRTLAST;MRTIVIFPi;VSSPi;CLTAUi;CENDINFi;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;FLGACCEPTTAU;THALFF"

> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 12596809
In addition: Warning message:
In run_M3_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 12596809 
```

[tc103_B2901001_M3SD.zip](https://github.com/tensfeldt/openNCA/files/3927304/tc103_B2901001_M3SD.zip)

**Comment by jhhughes256** [comment]
### Update

I can replicate this issue for tc008 M1SS & SD, tc010, tc011, tc012, tc013, tc014, tc015 and tc016 when using the PARAMETERDISPLAYLIST mentioned above (and provided below).

```
"CMAX;KEL;KELNOPT;KELRSQ;KELTMHI;KELTMLO;THALF"
```

Note: this error does not occur for the standard PARAMETERDISPLAYLIST for tc008 M1SS (#81) & SD, tc010, tc011, tc012, tc013, tc014, tc015 and tc016.

**Comment by jhhughes256** [comment]
### Update

For tc016 M3SS no failure.

```
mct$PARAMETERDISPLAYLIST <- "SUBJID;TREATXT;PKBDFLD;PKTERM;PERIODU;PERIOD;VISITU;VISIT;DOSE;DOSEU;ACTTRT;AUCTAUi;AUCLASTi;AUCXPCTP;CMAXi;CAVi;KEL;KELRSQ;KELNOPT;KELRSQ;KELTMHI;KELTMLO;THALF;TLAST;TMAXi;MRTLAST;MRTIVIFPi;VSSPi;CLTAUi;CENDINFi;THALFF"
```

For tc016 M3SS failure with XpctO

```
mct$PARAMETERDISPLAYLIST <- "SUBJID;TREATXT;PKBDFLD;PKTERM;PERIODU;PERIOD;VISITU;VISIT;DOSE;DOSEU;ACTTRT;AUCTAUi;AUCLASTi;AUCXPCTP;CMAXi;CAVi;KEL;KELRSQ;KELNOPT;KELRSQ;KELTMHI;KELTMLO;THALF;TLAST;TMAXi;MRTLAST;MRTIVIFPi;VSSPi;CLTAUi;CENDINFi;THALFF;FLGACCEPTKEL"
```

**Comment by rudraya** [comment]
This issue was addressed in fce7a3c, please let me know if it still occurs.

**Comment by jhhughes256** [comment]
### Update

These specific errors no longer occur as of fce7a3c. The errors being referred to are auc_XpctO and cest.
